### PR TITLE
add resource purchase_reserved_instances_offering

### PR DESCRIPTION
**Pull request by 956081712**
add resource purchase_reserved_instances_offering

### CODE REVIEW

File: alicloud/service_alicloud_ecs.go [hunk 1]
```diff
@@ -1035,6 +1035,47 @@ func (s *EcsService) DescribeSnapshotPolicy(id string) (*ecs.AutoSnapshotPolicy,
...
+func (s *EcsService) DescribePurchaseReservedInstanceOffering(id string) (reservedInstanc ecs.ReservedInstance, err error) {
+	request := ecs.CreateDescribeReservedInstancesRequest()
+	var balance = &[]string{id}
+	request.ReservedInstanceId = balance
+	request.RegionId = s.client.RegionId
+	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DescribeReservedInstances(request)
+	})
+	if err != nil {
+		return
```
Comment (Lexsss) [inline alicloud/service_alicloud_ecs.go]: nil, wrapErrorf()

File: alicloud/service_alicloud_ecs.go [hunk 2]
```diff
@@ -1035,6 +1035,47 @@ func (s *EcsService) DescribeSnapshotPolicy(id string) (*ecs.AutoSnapshotPolicy,
...
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	resp, _ := raw.(*ecs.DescribeReservedInstancesResponse)
```
Comment (Lexsss) [inline alicloud/service_alicloud_ecs.go]: response

File: website/docs/r/purchase_reserved_instance_offering.htm.markdown [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* `Period` - (Optional, ForceNew) Term of the RI. Unit: years. Optional values: 1 and 3.
+* `period_unit` - (Optional, ForceNew) Term unit. Optional value: Year.
+* `resource_group_id` - (Optional, ForceNew) Resource group ID.
+* `description` - (Optional) Description of the RI. 2 to 256 English or Chinese characters. It cannot start with http:// or https://.
+* `name` - (Optional) Name of the RI. The name must be a string of 2 to 128 characters in length and can contain letters, numbers, colons (:), underscores (_), and hyphens. It must start with a letter. It cannot start with http:// or https://.
+* `platform` - (Optional, ForceNew) The operating system type of the image used by the instance. Optional values: `Windows`, `Linux`. Default is `Linux`.
+
+### Timeouts
+
+-> **NOTE:** Available in 1.54.0+.
```
Comment (Lexsss) [inline website/docs/r/purchase_reserved_instance_offering.htm.markdown]: remove it

File: alicloud/provider.go [hunk 4]
```diff
@@ -238,46 +238,47 @@ func Provider() terraform.ResourceProvider {
...
-			"alicloud_ess_scalinggroup_vserver_groups":    resourceAlicloudEssScalingGroupVserverGroups(),
-			"alicloud_vpc":                                resourceAliyunVpc(),
-			"alicloud_nat_gateway":                        resourceAliyunNatGateway(),
-			"alicloud_nas_file_system":                    resourceAlicloudNasFileSystem(),
-			"alicloud_nas_mount_target":                   resourceAlicloudNasMountTarget(),
-			"alicloud_nas_access_group":                   resourceAlicloudNasAccessGroup(),
-			"alicloud_nas_access_rule":                    resourceAlicloudNasAccessRule(),
+			"alicloud_instance": resourceAliyunInstance(),
+			"alicloud_image":    resourceAliCloudImage(),
+			"alicloud_purchase_reserved_instance_offering": resourceAliCloudPurchaseReservedInstanceOffering(),
```
Comment (xiaozhu36) [inline alicloud/provider.go]: `alicloud_reserved_instance`

File: alicloud/resource_alicloud_purchase_reserved_instance_offering.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+	return &schema.Resource{
+		Create: resourceAliCloudPurchaseReservedInstanceOfferingCreate,
+		Read:   resourceAliCloudPurchaseReservedInstanceOfferingRead,
+		Update: resourceAliCloudPurchaseReservedInstanceOfferingUpdate,
+		Delete: resourceAliCloudPurchaseReservedInstanceOfferingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_purchase_reserved_instance_offering.go]: There is no need to set timeouts when timeout is less than 5 min.

File: website/docs/r/purchase_reserved_instance_offering.html.markdown [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  scope              ="Zone"
+  period             ="1"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `offering_type` - (Required, ForceNew) 	Payment type of the RI. Optional values: `No Upfront`: No upfront payment is required., `Partial Upfront`: A portion of upfront payment is required.`All Upfront`: Full upfront payment is required.
```
Comment (xiaozhu36) [inline website/docs/r/purchase_reserved_instance_offering.html.markdown]: Please retain one blank after （）

**Pull request merged**